Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp

```diff
@@ -1239,6 +1345,137 @@ WebIDL::ExceptionOr<void> HTMLInputElement::set_value_as_number(double value)
     return {};
 }
 
+// https://html.spec.whatwg.org/multipage/input.html#dom-input-stepup
+WebIDL::ExceptionOr<void> HTMLInputElement::step_up(long n)
```

Comment:
The spec states "The stepDown(n) and stepUp(n) methods, when invoked, must run the following algorithm:"

Let's not copy-paste the large function body for each of these, extract it to a common helper like `step_up_or_down` that `step_up` and `step_down` just invoke

Reply:
Good point, I need to stop with copying large parts of code and think a little bit more out of the box 😉